pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -684,8 +678,26 @@ func allocatableResources(memory, cpu string) v1.ResourceList {
 	}
 }
 
-// DaemonSets should not place onto nodes with insufficient free resource
+// When ScheduleDaemonSetPods is disabled, DaemonSets should not be placed onto nodes with insufficient free resource
+// We don't need this test as default scheduler is responsible for doing resource checks when ScheduleDaemonSetPods
+// is enabled. We are unsetting ScheduleDaemonSetPods to test codepath without ScheduleDaemonSetPods
```

Comment:
Please remove L682-683. We need this test for when the feature is disabled. It's a valid path since cluster admins can disable this feature. 

This comment applies globally.

Reply:
That is what I meant in this line `We are unsetting ScheduleDaemonSetPods to test...`. IIUC, once we graduate this feature to GA(v1), we will remove the code path specific to predicate checking from daemon_controller.go(current behaviour).

Reply:
This test makes sure it passes when the feature is disabled (non-default behavior). "We don't need this test" is misleading, and the rest of L682-683 is already covered by the first sentence. Also, "DaemonSets should not be placed" should be "DaemonSets should not place".